Sprint-1/fix/median.js

```diff
@@ -11,18 +11,33 @@
 // }
 
 // Explanation:
-// .splice() is a mutating method, so it updates the array in place.
-// to prevent this, you could take a copy of the array and use .splice(),
-// or you could use bracket notation to element at the desired index
+// The original implementation has several mistakes:
+// 1 - .splice() mutates or changes the original array. We don't want this to happen.
+// 2 - We actually don't need .splice() or any other method to find a median.
+// 3 - We must sort the array before we search for the median.
+// 4 - We should check if the 'list' parameter actually has numbers, otherwise try to filter
+// out all non-numeric values and throw an error or return null if there are no numeric values.
 
-function calculateMedian(list) {
-  const middleIndex = Math.floor(list.length / 2);
 
-  if (list.length % 2 === 0) { // use the remainder operator (%) to check if the array length is even
-    return (list[middleIndex - 1] + list[middleIndex]) / 2;
+// Example of a function calculating median and returning null in case no numeric values provided.
+
+function calculateMedian(list) {
+  if (!Array.isArray(list)) {
+    return null;
   }
+  const onlyNumberList = list.filter(item => typeof item === 'number' && !isNaN(item));
 
-  return list[middleIndex];
+  if (!onlyNumberList.length) {
+    return null;
+  } else {
+    const sortedList = onlyNumberList.sort((current, next) => current - next);
```

Comment:
The default comparator should work for numbers, right?

```suggestion
    const sortedList = onlyNumberList.sort();
```

Reply:
Not really, based on what I know about sort() if no comparator function is provided, it will convert numbers to strings and sort them alphabetically. Which will produce `wrong` results.

`console.log([10, 2, 1, 5].sort()); // [1, 10, 2, 5]`

Reply:
Oh wow, I knew the default comparator wasn't great, but I didn't realise it was _that_ not great!!